File: src/tests/fixtures/typescript/widget.ts
// Fixture: a small, domain-neutral TypeScript file exercising the kinds the
// plugin extracts (class, interface→trait, type→struct, enum, method, function,
// and an arrow assigned to a const).

export interface Renderer {
  render(): string;
}

export type WidgetSize = {
  width: number;
  height: number;
};

export enum WidgetColor {
  Red,
  Green,
}

export class Widget implements Renderer {
  constructor(private size: WidgetSize) {}

  render(): string {
    return `${this.size.width}x${this.size.height}`;
  }

  resize(width: number): void {
    this.size = { ...this.size, width };
  }
}

export function buildWidget(): Widget {
  return new Widget({ width: 1, height: 1 });
}

export const defaultWidget = () => buildWidget();

File: src/tests/fixtures/rust/widget.rs
//! Fixture: a small, domain-neutral Rust file exercising every kind the plugin
//! extracts (struct, enum, trait, free fn, impl methods, trait method).

pub struct Widget {
    pub size: u32,
}

pub enum Shape {
    Square,
    Round,
}

pub trait Render {
    fn render(&self) -> String;
}

impl Render for Widget {
    fn render(&self) -> String {
        String::new()
    }
}

impl Widget {
    pub fn new(size: u32) -> Self {
        Widget { size }
    }

    pub fn resize(&mut self, size: u32) {
        self.size = size;
    }
}

pub fn build_widget() -> Widget {
    Widget::new(0)
}
